src/blockchain/block.ts

```diff
@@ -145,50 +142,49 @@ export class Block {
 
   get registry(): Promise<TypeRegistry> {
     if (!this.#registry) {
-      this.#registry = Promise.all([this.metadata, this.#chain.api.chainProperties]).then(([data, properties]) => {
+      this.#registry = Promise.all([
+        this.metadata,
+        this.#chain.api.chainProperties,
+        this.#chain.api.chain,
+        this.runtimeVersion,
+      ]).then(([data, properties, chain, version]) => {
         const registry = new TypeRegistry(this.hash)
-        registry.setMetadata(new Metadata(registry, data))
         registry.setChainProperties(registry.createType('ChainProperties', properties))
+        registry.register(getSpecTypes(registry, chain, version.specName, version.specVersion))
+        registry.setHasher(getSpecHasher(registry, chain, version.specName))
+        registry.setMetadata(
+          new Metadata(registry, data),
+          undefined,
+          objectSpread<ExtDef>({}, getSpecExtensions(registry, chain, version.specName), {})
```

Comment:
I am still not 100% sure why use `objectSpread` instead of the actual spread syntax

Reply:
Not need. Replaced the rest with empty object and forgot to remove it